internal/common/schemafunc/find_changes.go

```diff
@@ -0,0 +1,201 @@
+package schemafunc
+
+import (
+	"context"
+	"fmt"
+	"slices"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+type AttributeChanges struct {
+	Changes []string
+}
+
+func (a *AttributeChanges) LeafChanges() map[string]bool {
+	return a.leafChanges(true)
+}
+
+func (a *AttributeChanges) AttributeChanged(name string) bool {
+	changes := a.LeafChanges()
+	changed := changes[name]
+	return changed
+}
+
+func (a *AttributeChanges) KeepUnknown(attributeEffectedMapping map[string][]string) []string {
+	keepUnknown := []string{}
```

Comment:
```suggestion
	var keepUnknown []string
```

nit: you can keep the var nil, and append will create it

Reply:
Thanks. Changed in 7dc03d07da78dda5733625fb913e89d5dfd67f82